feat: additional tracking

sshanzel opened the pull request:
## Changes
- Additional analytics events:
  - Scroll event for the Plus page. 
  - Hover plus feature. This means, wherever we use the component, the analytics events are sent even when not in Plus page.
  - Click plus FAQ. Same as above on when the event can be triggered.
  - Hover checkout widget. Same as above on when the event can be triggered.

## Events

Did you introduce any new tracking events?

<!--
If yes please remove the comment HTML comment tags and fill the table below

Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

-->

## Experiment

Did you introduce any new experiments?

<!--
If yes please remove the comment HTML comment tags and follow the instructions below

Don't forget to send a message to the [#experiments](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715175315620999) channel, following the template in slack, and adding a link to the message here.

> [!IMPORTANT]
> Please do not merge the PR until the experiment enrolment is approved.

-->

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

<!--
If relevant, please remove the comment HTML comment tags and fill the checkboxes below

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

-->

MI-863 #done

idoshamun commented:
@sshanzel i think Chris already implemented "Hover plus feature" for an experiment we had. Can you validate please?

sshanzel commented on packages/webapp/pages/plus/index.tsx:
```diff
@@ -45,9 +47,27 @@ const PlusPage = ({
+      });
+    };
+
+    globalThis?.window?.addEventListener('scrollend', onScroll);
+
+    return () => {
+      globalThis?.window?.removeEventListener('scrollend', onScroll);
```
We have to monitor when the scrolling ends rather than `scroll` event. Otherwise, we will get flooded by data with just a single scroll like below:

https://github.com/user-attachments/assets/4a4fe117-66aa-472d-8d27-087832c56ded

omBratteng replied:
Maybe some debouncing too?

sshanzel replied:
For the scrollends it is much straightforward. I thought of using debounce too but wasn't sure of what should be the right value.

sshanzel replied:
For the `scrollend` - I think debounce shouldn't be needed as seen below.

https://github.com/user-attachments/assets/41905962-bc39-47fa-b5d3-974718aec2a9

idoshamun reviewed (approved):
Only check my comment. Looks great!

sshanzel commented:
> @sshanzel i think Chris already implemented "Hover plus feature" for an experiment we had. Can you validate please?

Yep, seems to have existed from the container element. Removed what I added now 🚀

omBratteng commented on packages/shared/src/components/accordion/index.tsx:
```diff
@@ -7,21 +7,34 @@ import { ArrowIcon } from '../icons';
   const contentId = `accordion-content-${id}`;
 
+  const handleClick: MouseEventHandler<HTMLButtonElement> = (e) => {
+    if (onClick) {
+      onClick(e);
+    }
```
```suggestion
    onClick?.(e);
```

Is also an option

sshanzel replied:
Yeah, we used to do this, but way way back, it was said that curly braces were preferred. Although I don't mind using the single-liner.
